modules/iam-role-for-service-accounts-eks/policies.tf

```diff
@@ -302,6 +302,74 @@ resource "aws_iam_role_policy_attachment" "ebs_csi" {
   policy_arn = aws_iam_policy.ebs_csi[0].arn
 }
 
+################################################################################
+# FSx for Lustre CSI Driver Policy
+################################################################################
+
+# https://github.com/kubernetes-sigs/aws-fsx-csi-driver/blob/master/docs/README.md
+data "aws_iam_policy_document" "fsx_lustre_csi" {
+  count = var.create_role && var.attach_fsx_lustre_csi_policy ? 1 : 0
+
+  statement {
+    actions = [
+      "iam:CreateServiceLinkedRole",
+      "iam:AttachRolePolicy",
+      "iam:PutRolePolicy"
+    ]
+    resources = [
+      "arn:${local.partition}:iam::*:role/aws-service-role/s3.data-source.lustre.fsx.${local.dns_suffix}/*",
+    ]
+  }
+
+  statement {
+    actions = [
+      "iam:CreateServiceLinkedRole",
+    ]
+    resources = ["*"]
+    condition {
+      test     = "StringLike"
+      variable = "iam:AWSServiceName"
+      values   = ["fsx.${local.dns_suffix}"]
+    }
+  }
+
+  statement {
```

Comment:
I guess I overlooked this before? I don't remember it - but this is not in the policy so we can remove this whole statement block